scripts/nginx/readarr.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# Nginx conf for Radarr v3
+# Flying sausages 2020
+# Refactored by Bakerboy448 2021
+master=$(cut -d: -f1 < /root/.master.info)
+app_name="readarr"
+if ! READARR_OWNER="$(swizdb get $app_name/owner)"; then
+    READARR_OWNER=$(_get_master_username)
+else
+    READARR_OWNER="$(swizdb get $app_name/owner)"
+fi
+app_port="8787"
+user="$READARR_OWNER"
+app_servicename="${app_name}"
+app_configdir="/home/$user/.config/${app_name^}"
+app_baseurl=$app_name
+
+cat > /etc/nginx/apps/$app_name.conf << READARR
+location /$app_name {
+  proxy_pass        http://127.0.0.1:$app_port/;
+  proxy_set_header Host \$proxy_host;
+  proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
+  proxy_set_header X-Forwarded-Proto \$scheme;
+  proxy_redirect off;
+  auth_basic "What's the password?";
+  auth_basic_user_file /etc/htpasswd.d/htpasswd.${master};
+
+  proxy_http_version 1.1;
+  proxy_set_header Upgrade \$http_upgrade;
+  proxy_set_header Connection \$http_connection;
+  # Allow the App API
+  location /$app_baseurl/api { auth_request off;
+    proxy_pass http://127.0.0.1:$app_port/$app_baseurl/api;
+  }
+}
+READARR
+
+isactive=$(systemctl is-active $app_servicename)
+
+if [[ $isactive == "active" ]]; then
+    echo_log_only "Stopping $app_servicename"
+    systemctl stop $app_servicename
+fi
+user=$(grep User /etc/systemd/system/$app_servicename.service | cut -d= -f2)
+echo_log_only "${app_name^} user detected as $user"
+apikey=$(grep -oPm1 "(?<=<ApiKey>)[^<]+" "$app_configdir"/config.xml)
+echo_log_only "Apikey = $apikey" >> "$log"
+
+# Set to Debug as this is alpha software
+cat > "$app_configdir"/config.xml << READARR
+<Config>
+  <LogLevel>debug</LogLevel>
+  <UpdateMechanism>BuiltIn</UpdateMechanism>
+  <BindAddress>127.0.0.1</BindAddress>
+  <Port>$app_port</Port>
+  <SslPort>9898</SslPort>
+  <EnableSsl>False</EnableSsl>
+  <LaunchBrowser>False</LaunchBrowser>
+  <ApiKey>${apikey}</ApiKey>
+  <AuthenticationMethod>None</AuthenticationMethod>
+  <UrlBase>$app_baseurl</UrlBase>
+</Config>
+READARR
```

Comment:
We're only adding `UrlBase` here if I'm not mistaken. It might be worth seeing if this would be "simpler" to do via something like `jq` for XML, or via the API.